From c63990a2b0c5b0deb0fe7a1447a5f9a9d6818078 Mon Sep 17 00:00:00 2001
From: Zidan Wang <zidan.wang@freescale.com>
Date: Tue, 1 Dec 2015 16:14:26 +0800
Subject: [PATCH 0754/5242] MLK-11933 mfd: mxc-hdmi-core: null check substream
 before using it

commit  131e7697fba08a7fa54029763f0a80b30e1beb4a from
https://source.codeaurora.org/external/imx/linux-imx.git

null check substream before using it. Reported by coverity.

Signed-off-by: Zidan Wang <zidan.wang@freescale.com>
(cherry picked from commit fad2b9c1285a34eab0dec9f90edc06ec94e70a61)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mfd/mxc-hdmi-core.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/mfd/mxc-hdmi-core.c b/drivers/mfd/mxc-hdmi-core.c
index ca79e7e..a670406 100644
--- a/drivers/mfd/mxc-hdmi-core.c
+++ b/drivers/mfd/mxc-hdmi-core.c
@@ -153,9 +153,12 @@ int mxc_hdmi_register_audio(struct snd_pcm_substream *substream)
 	unsigned long flags, flags1;
 	int ret = 0;
 
+	if (!substream)
+		return -EINVAL;
+
 	snd_pcm_stream_lock_irqsave(substream, flags);
 
-	if (substream && check_hdmi_state()) {
+	if (check_hdmi_state()) {
 		spin_lock_irqsave(&hdmi_audio_lock, flags1);
 		if (hdmi_audio_stream_playback) {
 			pr_err("%s unconsist hdmi auido stream!\n", __func__);
-- 
1.7.9.5

